src/modules/ModuleInsights3Up/ModuleInsights3Up.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import { readFragment, ResultOf } from "gql.tada";
+import { GetModuleInsights3UpById } from "./GetModuleInsights3UpById";
+import { ComponentLinkFragment } from "@/lib/contentful/fragments/ComponentLinkFragment";
+import { Section } from "@/components/core/Section";
+import { Button } from "@/components/core/Button";
+import { Link } from "@/components/core/Link";
+import { Card } from "@/components/core/Card/Card";
+import { ScrollCarouselContainer } from "@/constants/ScrollCarouselContainer";
+import { getInspector } from "@/utils/inspectorMode";
+
+export const ModuleInsights3Up = ({
+  data,
+}: {
+  data: ResultOf<typeof GetModuleInsights3UpById>;
+}) => {
+  if (!data.moduleInsights3Up) return null;
+
+  const { headline, insightsCollection, cta } = data.moduleInsights3Up;
+
+  const inspector = getInspector(data.moduleInsights3Up);
+
+  const link = readFragment(ComponentLinkFragment, cta);
+  const insights = insightsCollection?.items.filter(Boolean) || [];
+
+  return (
+    <Section data-testid="ModuleInsights3Up" className="dark py-16 dsk:py-20">
+      {/* Header section with headline and CTA */}
+      <div className="flex flex-col dsk:flex-row dsk:items-center dsk:justify-between dsk:mb-16 gap-10">
+        <h2
+          className="typo-heading-2 font-light max-w-2xl"
+          {...inspector("headline")}
+        >
+          {headline}
+        </h2>
+
+        {link && (
+          <Button asChild>
+            <Link link={link} {...inspector("cta")}>
+              {link.label}
+            </Link>
+          </Button>
+        )}
```

Comment:
```suggestion
        {data.moduleInsights3Up.cta && (
          <Button asChild>
            <Link link={data.moduleInsights3Up.cta} {...inspector("cta")} />
          </Button>
        )}
```

You don't need to deconstruct these fragments; I've set the component up to be able to accept the wrapped or unwrapped fragment. And the label should pass as the children automatically